search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAParameters.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Defines parameters for generative QA search pipelines.
+ *
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+public class GenerativeQAParameters implements Writeable, ToXContentObject {
+
+    private static final ObjectParser<GenerativeQAParameters, Void> PARSER;
+
+    private static final ParseField CONVERSATION_ID = new ParseField("conversation_id");
+    private static final ParseField LLM_MODEL = new ParseField("llm_model");
+    private static final ParseField LLM_QUESTION = new ParseField("llm_question");
+
+    static {
+        PARSER = new ObjectParser<>("generative_qa_parameters", GenerativeQAParameters::new);
+        PARSER.declareString(GenerativeQAParameters::setConversationId, CONVERSATION_ID);
+        PARSER.declareString(GenerativeQAParameters::setLlmModel, LLM_MODEL);
+        PARSER.declareString(GenerativeQAParameters::setLlmQuestion, LLM_QUESTION);
+    }
+
+    @Setter
+    @Getter
+    private String conversationId;
+
+    @Setter
+    @Getter
+    private String llmModel;
+
+    @Setter
+    @Getter
+    private String llmQuestion;
+
+    public GenerativeQAParameters(StreamInput input) throws IOException {
+        this.conversationId = input.readOptionalString();
+        this.llmModel = input.readOptionalString();
+        this.llmQuestion = input.readString();
```

Comment:
It's optional because you can specify an OpenSearch `model_id` (or an alias #1217) which allows you to create an HttpConnector with a template that can contain a default model (e.g. gpt-3.5-turbo).